common/lib/capa/capa/safe_exec/safe_exec.py

```diff
@@ -143,28 +144,41 @@ def safe_exec(
     # Create the complete code we'll run.
     code_prolog = CODE_PROLOG % random_seed
 
-    # Decide which code executor to use.
-    if unsafely:
-        exec_fn = codejail_not_safe_exec
-    else:
-        exec_fn = codejail_safe_exec
-
-    # Run the code!  Results are side effects in globals_dict.
-    try:
-        exec_fn(
-            code_prolog + LAZY_IMPORTS + code,
-            globals_dict,
-            python_path=python_path,
-            extra_files=extra_files,
-            limit_overrides_context=limit_overrides_context,
-            slug=slug,
-        )
-    except SafeExecException as e:
-        # Saving SafeExecException e in exception to be used later.
-        exception = e
-        emsg = text_type(e)
+    if is_codejail_rest_service_enabled():
+        data = {
+            "code": code_prolog + LAZY_IMPORTS + code,
+            "globals_dict": globals_dict,
+            "python_path": python_path,
+            "limit_overrides_context": limit_overrides_context,
+            "slug": slug,
+            "unsafely": unsafely
+        }
+
+        emsg, exception = send_safe_exec_request(data, extra_files)
```

Comment:
since we are already talking about a next version of this service that might cache the extra_files I think I'd like to raise a point here. It does not have to be blocking given that we also want this for maple so that tutor installs can use codejail.

We could call this version `send_safe_exec_request_v0` and make the calling here configurable.
Then this _v0 function will be tied to the `v0` url path in the remote API call.

Once we start upgrading the service and the call here to use v1 with the cache and whatnot we dont need to necessarily break the usage for those that do not want to install a more complicated codejail service or simply for those that need to upgrade step by step and might need to leave the v0 service running for a bit longer.

If we want to make it easy, I would say we only need to make one function signature for v0 and subsequent, for instance by adding the `extra_files` to the `data` dict and pass just that. Also we need to rely not only in is_codejail_rest_service_enabled() but also one function that does `get_remote_exec function` that returns send_safe_exec_request_v0 for now. Once we are ready to extend we only need to extend the `remote` or `helpers` module to add support for v1 while v0 will still be usable.